assets/js/components/_Dismissible.js

```diff
@@ -0,0 +1,69 @@
+'use strict';
+
+const utils = require('../libs/elife-utils')();
+
+module.exports = class Dismissible {
+  constructor($toDismiss, attachPoint, document) {
+    this.$toDismiss = $toDismiss;
+    this.doc = document;
+
+    this.cookieName = Dismissible.deriveCookieName(this.$toDismiss);
+    if (this.hasCookieName(this.cookieName) && this.hasPreviouslyBeenDismissed(this.cookieName, this.doc.cookie)) {
+      this.hide();
+      return;
+    }
+
+    this.cookieExpiryDate = Dismissible.setCookieExpiryDate(this.$toDismiss);
+    this.$button = this.buildButton(attachPoint);
+  }
+
+  static setCookieExpiryDate($elm) {
```

Comment:
a `set*` that returns rather than modifying `this` is a bit unexpected, but maybe I'm not aware of a naming convention in use here

Reply:
(for now I'll just leave it)